api/envoy/extensions/matching/input_matchers/ip/v3/ip.proto

```diff
@@ -23,6 +23,7 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // <https://www.nada.kth.se/~snilsson/publications/IP-address-lookup-using-LC-tries/>`_
 // by S. Nilsson and G. Karlsson. For "big" lists of IPs, this matcher is more
 // efficient than multiple single IP matcher, that would have a linear cost.
+// Envoy's LC-trie implementation supports up to 262144 unique entries.
```

Comment:
Thanks for the update. For my understanding do you know why this limitation is the way it is? Can you potentially say a bit more for the next person? Thank you.

/wait

Reply:
Sure - my understanding based on a quick code dive is that this limit is a result of the address size used for each node in the LC-Trie (20 bits) combined with the default `fill_factor` of 0.5. This means the maximum number of populated nodes is 2 ^ 20 * 0.5 = 2 ^ 19. However, step 2 in the algorithm for populating the LC-Trie can cause up to 2 * N elements to be populated where N is the number of elements in the input. Thus in practice the maximum number of elements that can be supplied to the Trie by the IP matcher / IP tagging filter is 2 ^ 18 = 262144.

[Relevant code comment](https://github.com/envoyproxy/envoy/blob/81359e48a6d965d2b3630b40bf94fe074d44fa69/source/common/network/lc_trie.h#L61-L65)